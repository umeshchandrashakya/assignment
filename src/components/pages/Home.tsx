import * as React from "react";

import { Link } from "react-router-dom";

class Home extends React.Component {
  render() {
    return (
      <div>
        <nav>
          <Link id='login' to="/login">Login </Link>
          <br></br>
          <Link id='reg' to="/registration">Registration</Link>
          <br></br>
          <br></br>
        </nav>
      </div>
    );
  }
}

export default Home;
