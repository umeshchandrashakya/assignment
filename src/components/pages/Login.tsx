import React from "react";
import { useActions } from "typeless";
import { LoginActions, getLoginState } from "../typeless/interface/login";

const Login: React.FC = () => {
  // wrap actions with `dispatch`
  const { usernameChange, passwordChange, signIn } = useActions(LoginActions);
  // get state from store
  const { username, password, isLoading, error, isSignedIn, formSubmitStatus } = getLoginState.useState();

  const onLoginClicked = () => {
    signIn(username, password);
  }

  const getStatus = () => {
     if (formSubmitStatus) {
       return isLoading ? 'Loading, please wait...' : (isSignedIn ? 'Successfully logged in' : `Error: ${error}`)
     }
     return '';
  }

  return (
    <div>
      <h1>Login</h1>
      <form>
        <input className='input1' value={username} placeholder='Username' onChange={e => usernameChange(e.target.value)} />
        <input className='input2' value={password} placeholder='Password' onChange={e => passwordChange(e.target.value)} />
        <button type='button' onClick={onLoginClicked}>Login</button>
      </form>
      <h3>{getStatus()}</h3>
    </div>
  );
};

export default Login;
