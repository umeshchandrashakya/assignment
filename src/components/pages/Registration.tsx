import React from "react";
import Field from "../molecules/Field";
import Label from "../atoms/Label";
import styled from "styled-components";
import Button from "../atoms/Button";
import image from "../../assests/image.jpg";
import image1 from "../../assests/image1.jpg";
import { useActions } from "typeless";
import { RegActions, getRegState } from "../typeless/interface/reg";
import { url } from "inspector";
import RegForm from "../template/RegForm";

const StyledRegistration = styled.div`
 
  display: flex;
  flex-flow: column;
  margin: 0 auto;
  border: 1px ;
  background-repeat: no-repeat;
  background-image: url(${image});
  height: 140vh;
  width: 100vw;
  background-size: cover;
  background-position: center center;
  box-sizing: border-box;
`;



const Registration: React.FC = () => {

 return (
  
  <StyledRegistration >
    
     
    <RegForm title="REGISTRATION"></RegForm>
    
   
      </StyledRegistration>
      
  
  );
};

 export default Registration;
