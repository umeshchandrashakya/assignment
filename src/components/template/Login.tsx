import React from "react";
import styled from "styled-components";
import InputText from "../atoms/InputText";
import Label from "../atoms/Label";
import Field from "../molecules/Field";
import Form from "../organisms/LoginForm";
import img3 from "../../assets/img3.jpg";
import { StyledSpinner } from "../atoms/StyledSpinner";

const StyledLogin = styled.div`
  display: flex;
  align-items: center;
  flex-flow: column;
  /* width: 350px;
  height: 350px; */
  margin: 0 auto;
  background-repeat: no-repeat;
  background-image: url(${img3});
  height: 100vh;
  width: 100vw;
  background-size: cover;
  background-position: center center;
  box-sizing: border-box;
  min-height: 30em;
  min-width: max-content;
  position: relative;
  /* min-width: 25em; */
  h2 {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 14px;
  }
  .spinner-container {
    width: 100vw;
    height: 100vh;
    z-index: 1;
    position: absolute;
    justify-content: center;
    display: flex;
    align-items: center;
  }
`;

interface Props {
  title: string;
  isLoading: boolean;
  onUsernameChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onPasswordChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  userNameEmpty: boolean;
  passwordEmpty: boolean;
  onSignupClick: (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void;
}

const LoginForm = (props: Props) => (
  <StyledLogin className="container-fluid">
    <div
      className="spinner-container"
      style={{ visibility: props.isLoading ? "visible" : "hidden" }}
    >
      <StyledSpinner></StyledSpinner>
    </div>
    <Form {...props} />
  </StyledLogin>
);

export default LoginForm;
