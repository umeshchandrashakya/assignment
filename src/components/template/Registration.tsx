import React from "react";
import styled from "styled-components";
import RegistrationForm from "../organisms/RegistrationForm";
import image from "../../assets/bg1.jpg";
import Header from '../organisms/Header';
import Footer from '../organisms/Footer';

const StyledRegistration = styled.div`
background-repeat: no-repeat;
background-image: url(${image});
background-size: cover;
background-attachment:fixed;
height:100vh;
overflow-y:scroll;
`;
interface Props {
    title: string;
    isSuccess?: boolean;
    isLoading?:boolean;
    validName?:boolean;
    validEmail?:boolean;
    validPhone?:boolean;
    validPassword?:boolean;
    onNameChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
    onEmailChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
    onPasswordChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
    onContactNoChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
    onRegisterClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
    onLoginClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const RegForm = (props: Props) => (
    <StyledRegistration className='row align-items-center justify-content-center'>
        <div className='col-sm'>
            <Header />
            <RegistrationForm {...props} />
            <Footer />
        </div>
    </StyledRegistration>
);
export default RegForm;


