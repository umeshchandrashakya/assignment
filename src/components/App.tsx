import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Login from "./pages/Login";
import Registration from "./pages/Registration";
import Home from "./pages/Home";
import img from "../img/sample.jpg";
const StyledLogin = styled.div`
  background-repeat: no-repeat;
  background-image: url(${img});
  height: 100vh;
  width: 100vw;
  background-size: cover;
  background-position: center center;
  box-sizing: border-box;
`;

const App: React.FC = () => {
  return (
    <StyledLogin>
      <Router>
        <div>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/login" component={Login} />
            <Route target exact path="/registration" component={Registration} />
          </Switch>
        </div>
      </Router>
    </StyledLogin>
  );
};

export default App;
