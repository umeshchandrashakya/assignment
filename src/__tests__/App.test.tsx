import * as React from "react";
import App from "../components/App";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount, render } from "enzyme";
import { MemoryRouter } from "react-router";
import Home from "../components/pages/Home";
import Login from "../components/pages/Login";
import Registration from "../components/pages/Registration";

describe("<App />", () => {
  it("renders without crashing", () => {
    const component = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>);
    expect(component).toMatchSnapshot();
  });

  test("invalid path should redirect to 404", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/random"]}>
          <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Home)).toHaveLength(0);
  });

  test("valid path should not redirect to Login", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/login"]}>
          <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Login)).toHaveLength(1);
  });
  test("valid path should redirect to Registration", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/registration"]}>
          <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Registration)).toHaveLength(1);
  });
  it("has a div tag", () => {
    const warpper = shallow(<App />);
    expect(warpper.find("div")).toHaveLength(1);
  });
});
