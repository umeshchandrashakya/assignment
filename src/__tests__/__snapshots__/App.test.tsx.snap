// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-htoDjs clirjG",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "attribs": Object {},
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "href": "/login",
                          "id": "login",
                        },
                        "children": Array [
                          Object {
                            "data": "Login ",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "a",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {},
                          "children": Array [],
                          "name": "br",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": Object {
                            "attribs": Object {
                              "href": "/registration",
                              "id": "reg",
                            },
                            "children": Array [
                              Object {
                                "data": "Registration",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "a",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": Object {
                              "attribs": Object {},
                              "children": Array [],
                              "name": "br",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": Object {
                                "attribs": Object {},
                                "children": Array [],
                                "name": "br",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": [Circular],
                                "type": "tag",
                                "x-attribsNamespace": Object {},
                                "x-attribsPrefix": Object {},
                              },
                              "parent": [Circular],
                              "prev": [Circular],
                              "type": "tag",
                              "x-attribsNamespace": Object {},
                              "x-attribsPrefix": Object {},
                            },
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "href": undefined,
                              "id": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "href": undefined,
                              "id": undefined,
                            },
                          },
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "href": undefined,
                          "id": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "href": undefined,
                          "id": undefined,
                        },
                      },
                      Object {
                        "attribs": Object {},
                        "children": Array [],
                        "name": "br",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {
                            "href": "/registration",
                            "id": "reg",
                          },
                          "children": Array [
                            Object {
                              "data": "Registration",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "a",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": Object {
                            "attribs": Object {},
                            "children": Array [],
                            "name": "br",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": Object {
                              "attribs": Object {},
                              "children": Array [],
                              "name": "br",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": [Circular],
                              "type": "tag",
                              "x-attribsNamespace": Object {},
                              "x-attribsPrefix": Object {},
                            },
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "href": undefined,
                            "id": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "href": undefined,
                            "id": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "href": "/login",
                            "id": "login",
                          },
                          "children": Array [
                            Object {
                              "data": "Login ",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "a",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "href": undefined,
                            "id": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "href": undefined,
                            "id": undefined,
                          },
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {},
                        "x-attribsPrefix": Object {},
                      },
                      Object {
                        "attribs": Object {
                          "href": "/registration",
                          "id": "reg",
                        },
                        "children": Array [
                          Object {
                            "data": "Registration",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "a",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {},
                          "children": Array [],
                          "name": "br",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": Object {
                            "attribs": Object {},
                            "children": Array [],
                            "name": "br",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {},
                          "children": Array [],
                          "name": "br",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {
                              "href": "/login",
                              "id": "login",
                            },
                            "children": Array [
                              Object {
                                "data": "Login ",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "a",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "href": undefined,
                              "id": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "href": undefined,
                              "id": undefined,
                            },
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "href": undefined,
                          "id": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "href": undefined,
                          "id": undefined,
                        },
                      },
                      Object {
                        "attribs": Object {},
                        "children": Array [],
                        "name": "br",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {},
                          "children": Array [],
                          "name": "br",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "href": "/registration",
                            "id": "reg",
                          },
                          "children": Array [
                            Object {
                              "data": "Registration",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "a",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {},
                            "children": Array [],
                            "name": "br",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": Object {
                              "attribs": Object {
                                "href": "/login",
                                "id": "login",
                              },
                              "children": Array [
                                Object {
                                  "data": "Login ",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "text",
                                },
                              ],
                              "name": "a",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": [Circular],
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "href": undefined,
                                "id": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "href": undefined,
                                "id": undefined,
                              },
                            },
                            "type": "tag",
                            "x-attribsNamespace": Object {},
                            "x-attribsPrefix": Object {},
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "href": undefined,
                            "id": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "href": undefined,
                            "id": undefined,
                          },
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {},
                        "x-attribsPrefix": Object {},
                      },
                      Object {
                        "attribs": Object {},
                        "children": Array [],
                        "name": "br",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {},
                          "children": Array [],
                          "name": "br",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {
                              "href": "/registration",
                              "id": "reg",
                            },
                            "children": Array [
                              Object {
                                "data": "Registration",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "a",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": Object {
                              "attribs": Object {},
                              "children": Array [],
                              "name": "br",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": [Circular],
                              "parent": [Circular],
                              "prev": Object {
                                "attribs": Object {
                                  "href": "/login",
                                  "id": "login",
                                },
                                "children": Array [
                                  Object {
                                    "data": "Login ",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "text",
                                  },
                                ],
                                "name": "a",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": [Circular],
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "href": undefined,
                                  "id": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "href": undefined,
                                  "id": undefined,
                                },
                              },
                              "type": "tag",
                              "x-attribsNamespace": Object {},
                              "x-attribsPrefix": Object {},
                            },
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "href": undefined,
                              "id": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "href": undefined,
                              "id": undefined,
                            },
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {},
                          "x-attribsPrefix": Object {},
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {},
                        "x-attribsPrefix": Object {},
                      },
                    ],
                    "name": "nav",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {},
                    "x-attribsPrefix": Object {},
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
